packages/core/admin/server/src/services/statistics.ts

```diff
@@ -0,0 +1,425 @@
+  username: string;
+  firstname: string;
+  lastname: string;
+}
+
+export interface Payload {
+  entry: {
+    documentId: string;
+    locale?: string;
+  };
+  uid: string;
+}
+
+export type Activity = {
+  action: string;
+  createdAt: string;
+  date: string;
+  documentId: string;
+  id: number;
+  locale: string;
+  payload: Payload;
+  publishedAt: string;
+  updatedAt: string;
+  user: User;
+};
+
+export type Release = {
+  createdAt: string;
+  documentId: string;
+  id: number;
+  locale: string;
+  name: string;
+  publishedAt: string;
+  releasedAt: string;
+  scheduledAt: string;
+  status: string;
+  timezone: string;
+  updatedAt: string;
+};
+
+export type EEStatistics = {
+  upcomingReleases: Release[] | Array<object>;
+  latestActivities: Activity[] | Array<object>;
+  upcomingReleasesCount: number,
+  workflowsCount: number,
+};
+
+const providers: Map<string, KeyNumberProvider> = new Map();
+
+export function registerKeyNumbersProvider(provider: KeyNumberProvider): void {
+  if (providers.has(provider.name)) {
+    throw new Error(`Provider with name ${provider.name} already exists`);
+  }
+
+  providers.set(provider.name, provider);
+}
+
+export async function getKeyNumbers(): Promise<KeyNumbers> {
+  const keyNumbers: KeyNumbers = {
+    entries: 0,
+    assets: 0,
+    contentTypes: 0,
+    components: 0,
+    locales: 0,
+    admin_users: 0,
+    webhooks: 0,
+    apiTokens: 0
+  };
+
+  const results = Array.from(providers.values()).map((provider) =>
+    provider.resolve().then((result) => {
+      keyNumbers[provider.name] = result;
+    })
+  );
+
+  await Promise.all(results);
+
+  return keyNumbers;
+}
+
+export async function getContentTypeStatistics(contentType: UID.CollectionType, userId: number): Promise<ContentTypeStatistics> {
+  // Fetch entries with associated user data that would always be there and just enough for the dashboard
+  const [entries, initialCount] = await strapi.db.query(contentType).findWithCount({
+    select: ['createdAt', 'updatedAt', 'publishedAt', 'locale', 'documentId'],
+    populate: {
+      createdBy: {
+        select: ['username', 'firstname', 'lastname']
+      },
+      updatedBy: {
+        select: ['username', 'firstname', 'lastname']
+      },
+      strapi_stage: {
+        select: ['name']
+      },
+      strapi_assignee: {
+        select: ['id']
+      }
+    },
+    orderBy: { publishedAt: 'DESC' }
+  });
+
+  let count = initialCount;
+
+  // Initialize data structures
+  const published: Entry[] = [];
+  const draft: Entry[] = [];
+  const locales: Locales = {};
+  const assignedEntries: Entry[] = [];
+
+  const contributors: Record<string, any> = {};
+  const stackedAreaChartByLocaleData: Record<string, any> = {};
+  const stackedAreaChartByStageData: Record<string, any> = {};
+  const stackedAreaChartByAssignedStageData: Record<string, any> = {};
+  const publishedDocumentIds = new Set<string>();
+
+  // Process each entry
+  entries.forEach(entry => {
```

Comment:
I think this may take a while on large databases, we need to ensure it's behaving well in such case like:

- Add a timeout, to not waste server resources in case it's very slow (like 10s)
- Ensure the frontend is not blocked when the data is loading
- Caching could help too, but it seems complex to achieve the right way